src/library_webgl.js

```diff
@@ -1813,7 +1813,7 @@ var LibraryGL = {
 #if MAX_WEBGL_VERSION >= 2
     if (GL.currentContext.version >= 2) { // WebGL 2 provides new garbage-free entry points to call to WebGL. Use those always when possible.
       if (data) {
-        GLctx.bufferData(target, HEAPU8, usage, data, size);
+        if (size) GLctx.bufferData(target, HEAPU8, usage, data, size);
```

Comment:
If one calls `bufferData` with zero size, I would expect that it should either:
 a) zero out the size of the buffer storage on the GPU (shrink the buffer to zero size and free the old GPU memory), or
 b) issue a GL error about programming bug. (and the old buffer storage size stays intact)

(not sure which currently happens by the spec)

Why should we turn either of behaviors a) or b) to silent no-op pass throughs? a) could be useful behavior, and b) would help with catching programming errors?

Reply:
This is a good point though; this is an implementation of glBufferData and passing 0 for the size there should actually reallocate and change the size of the underlying buffer. Skipping the call here means the old size will be preserved.

To fix this we would have to create and pass in an actually-zero-sized srcData.

Reply:
Fixed for bufferData, compressedTexImage2D, and compressedTexImage3D.